tests/jballerina-unit-test/src/test/java/org/ballerinalang/test/expressions/listconstructor/ListConstructorExprTest.java

```diff
@@ -61,10 +61,12 @@ public void diagnosticsTest() {
                 "inferred for '[1, p]'", 38, 35);
         BAssertUtil.validateError(resultNegative, i++, "invalid list constructor expression: types cannot be " +
                 "inferred for '[a, 4]'", 41, 23);
-        BAssertUtil.validateError(resultNegative, i++, "tuple and expression size does not match",
+        BAssertUtil.validateError(resultNegative, i++,
```

Comment:
In cases like
```ballerina
type Rec record {
  int a;
};

function foo() {
  [Rec1, int] a = []; //this should not throw no filler value error AFIU. 
}
```

There seems to be already this bug. The only difference is now we are getting this new error. 
Created https://github.com/ballerina-platform/ballerina-lang/issues/35712